### PR TITLE
Test Kokkos on GPU

### PR DESCRIPTION
**Pull request by zjiaqi2018**
Test using Kokkos on GPU, but only works when `CMAKE_BUILD_TYPE=Debug ` is set. @tjhei

### CODE REVIEW

**Comment by zjiaqi2018** [comment]
The script to build deal.ii
```
cmake  -D CMAKE_INSTALL_PREFIX=../installed-kokkos \
              -D CMAKE_BUILD_TYPE=Debug \
              -D CMAKE_CXX_COMPILER=/home/jiaqi2/kokkos/bin/nvcc_wrapper \
              -D DEAL_II_CUDA_FLAGS='-arch=sm_70' \
              -D DEAL_II_CXX_FLAGS='-Werror -Wno-non-template-friend' \
              -D DEAL_II_EARLY_DEPRECATIONS=ON \
              -D DEAL_II_WITH_CUDA="ON" \
              -D DEAL_II_WITH_KOKKOS="ON" \
              -D KOKKOS_DIR=/home/jiaqi2/kokkos/installed \
              -D DEAL_II_WITH_MPI="ON" \
              -D DEAL_II_WITH_P4EST="ON" \
              -D DEAL_II_WITH_TRILINOS="OFF" \
              -D DEAL_II_COMPONENT_EXAMPLES="OFF" \
	      ../dealii
```

File: tests/cuda/test_kokkos.cc [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    for (int i = 0; i < N; ++i)
+      {
+        h_y(i) = 1;
+      }
+
+    // Initialize x vector on host.
+    for (int i = 0; i < N; ++i)
+      {
+        h_x(i) = 1;
+      }
```
Comment (masterleinad) [inline tests/cuda/test_kokkos.cc:57]:
```suggestion
    // Initialize y vector on host.
    Kokkos::deep_copy(h_y, 1);
    
    // Initialize x vector on host.
    Kokkos::deep_copy(h_x, 1);
```

File: tests/cuda/test_kokkos.cc [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+  Kokkos::initialize(argc, argv);
+  {
+    using MemSpace = Kokkos::CudaSpace;
+
+    using ExecSpace = MemSpace::execution_space;
+
+    using range_policy = Kokkos::RangePolicy<ExecSpace>;
+
+    using ViewVectorType = Kokkos::View<double *, Kokkos::LayoutLeft, MemSpace>;
```
Comment (masterleinad) [inline tests/cuda/test_kokkos.cc:38]:
```suggestion
    using ExecSpace = MemSpace::execution_space;
    using range_policy = Kokkos::RangePolicy<ExecSpace>;

    using ViewVectorType = Kokkos::View<double *>;
```

File: tests/cuda/test_kokkos.cc [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+ * Public License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ * The full text of the license can be found in the file LICENSE.md at
+ * the top level directory of deal.II.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+
+// Compute the inner product of two vectors on the GPU using Kokkos.
```
Comment (masterleinad) [inline tests/cuda/test_kokkos.cc:18]:
```suggestion
// Compute the inner product of two vectors on the default execution space using Kokkos.
```

**Review by masterleinad — CHANGES_REQUESTED** [review]
```
 DEAL_II_CUDA_FLAGS='-arch=sm_70'
```
doesn't have any effect anymore. We only use flags coming from `Kokkos`.

Also, note that we already have a quick test which also tests basic functionality (and isn't limited to `CUDA`), see https://github.com/dealii/dealii/blob/master/tests/quick_tests/kokkos.cc. In general, I think we should go away from tests that only work with `CUDA` as opposed to using the default execution space in `Kokkos`.

Also, why would this only work in `Debug` mode?

**Comment by masterleinad** [comment]
Also, this doesn't use deal.II at all.

**Comment by zjiaqi2018** [comment]
> ```
>  DEAL_II_CUDA_FLAGS='-arch=sm_70'
> ```
> 
> doesn't have any effect anymore. We only use flags coming from `Kokkos`.

Should we remove it from workflows/linux.yml?
> 
> Also, note that we already have a quick test which also tests basic functionality (and isn't limited to `CUDA`), see https://github.com/dealii/dealii/blob/master/tests/quick_tests/kokkos.cc. In general, I think we should go away from tests that only work with `CUDA` as opposed to using the default execution space in `Kokkos`.
> 
Then this test is not necessary anymore. 

> Also, why would this only work in `Debug` mode?
It works in both release and debug modes after I rebase to master.

**Pull request closed without merging**